### PR TITLE
Add obs_location property to misfits and corresponding test

### LINKED ISSUE #1373: Expose obs_location (xaxis value) in addition to obs_index for misfits
Currently only `obs_index` is exposed in json for univariate misfits, which makes it difficult to place it right in the visualization plots. Therefore we can also directly expose its location on x-axis too.

### PR DESCRIPTION
**Pull request by xjules**
Resolves #1373

The PR includes the updated json data as as well as the corresponding test.

### CODE REVIEW

File: ert_shared/storage/endpoints/responses.py [hunk 1]
```diff
@@ -21,7 +21,12 @@ def _calculate_misfit(
     misfit = (difference / observation_std) ** 2
     sign = difference > 0
 
-    return {"value": misfit, "sign": sign, "obs_index": obs_index}
+    return {
+        "value": misfit,
+        "sign": sign,
+        "obs_index": obs_index,
```
Comment (oysteoh) [inline ert_shared/storage/endpoints/responses.py:27]: do we still need to send the observation index? As far as i remember it didn't make much sense in the gui? But i might remember wrong 🤔

Comment (xjules) [reply ert_shared/storage/endpoints/responses.py:27]: > do we still need to send the observation index? As far as i remember it didn't make much sense in the gui? But i might remember wrong

Decided to leave it there as we weren't sure :shrug: yesterday if it is allowed to two obs values with the same location; ie. `obs_index` separates them.

File: ert_shared/storage/endpoints/responses.py [hunk 2]
```diff
@@ -101,6 +106,7 @@ async def read_response_by_id(*, db: Session = Db(), ensemble_id: int, id: int):
             obs_values = list(observation.values)
             obs_stds = list(observation.errors)
             obs_data_indexes = list(observation.data_indices)
+            obs_key_indexes = list(observation.key_indices)
```
Comment (oysteoh) [inline ert_shared/storage/endpoints/responses.py:109]: Should this new key in the json also have a corresponding test ?

Comment (xjules) [reply ert_shared/storage/endpoints/responses.py:109]: Doesn't this https://github.com/equinor/ert/pull/1375/files#diff-dff280db3a36b0c8a92239a8f48346cc0178f5f00c02a0d0b360057544828e89 suffice or you had some other in mind?

Comment (oysteoh) [reply ert_shared/storage/endpoints/responses.py:109]: naaah, i was thinking more updating this test with the new property 😃 
```
def test_observation(app_client):
    ens = app_client.get(p.ensemble(1)).json()
    expected = {
        "active_mask": [True, False],
        "data_indexes": [2, 3],
        "key_indexes": [0, 3],
        "std": [1, 3],
        "values": [10.1, 10.2],
    }
...
...
```

Comment (oysteoh) [reply ert_shared/storage/endpoints/responses.py:109]: Forget it - i was fooled by the way github collapsed code! 👍  Looks good !

**Review by oysteoh — APPROVED** [review]
🚀

**Pull request merged**